TechnicalTask-lvl1/UsersListModule/UsersListViewModel/UsersListViewModel.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  UsersListViewModel.swift
+//  TechnicalTask-lvl1
+//
+//  Created by Сергей Матвеенко on 25.11.24.
+//
+
+import Foundation
+import Combine
+
+final class UsersListViewModel: UsersListViewModelProtocol {
+    
+    // MARK: - Parameters
+    
+    private let updatingUsersDataFacade: UpdatingUsersDataFacadeProtocol
+    private let coreDataService: CoreDataServiceProtocol
+    private var cancellables: Set<AnyCancellable> = []
+
+    var displayData = [UsersListDiplayModel]()
+    
+    private let loaderIsActivePublisher = PassthroughSubject<Bool, Never>()
+    var anyLoaderIsActivePublisher: AnyPublisher<Bool, Never> {
+        self.loaderIsActivePublisher.eraseToAnyPublisher()
+    }
+    
+    private let addButtonTappedPublisher = PassthroughSubject<Void, Never>()
+    var anyAddButtonTappedPublisher: AnyPublisher<Void, Never> {
+        self.addButtonTappedPublisher.eraseToAnyPublisher()
+    }
+    
+    private let displayDataUpdatedPublisher = PassthroughSubject<Void, Never>()
+    var anyDisplayDataUpdatedPublisherPublisher: AnyPublisher<Void, Never> {
+        self.displayDataUpdatedPublisher.eraseToAnyPublisher()
+    }
+    
+    private let isShowingInternetErrorPublisher = PassthroughSubject<Bool, Never>()
+    var anyIsShowingInternetErrorPublisher: AnyPublisher<Bool, Never> {
+        self.isShowingInternetErrorPublisher.eraseToAnyPublisher()
+    }
+    
+    // MARK: - Initialization
+    
+    init(updatingUsersDataFacade: UpdatingUsersDataFacadeProtocol, coreDataService: CoreDataServiceProtocol) {
+        self.updatingUsersDataFacade = updatingUsersDataFacade
+        self.coreDataService = coreDataService
+    }
+    
+    // MARK: - Events
+    
+    func readyToDisplay() {
```

Comment:
In my opinion, I would rename this method to something that better reflects the action it performs :)